lib/active_model/serializer.rb

```diff
@@ -182,7 +182,7 @@ def as_json(adapter_opts = nil)
 
     # Used by adapter as resource root.
     def json_key
```

Comment:
Thinking about it now, shouldn't it be the responsibility of the adapter to determine the json key/root?